### PR TITLE
Make use of auth customer

### PR DESCRIPTION
**Pull request by taha7**
***What***

1.  Fix the create customer seeder and generate a refresh token on create
2.  Load the whole customer data with the context
3. Authenticate the queries/mutations that requires auth 
4. Use the customer id in the according resolvers

***Links***

- [Trello Task](https://trello.com/c/7qPAyR3S/17-make-use-of-the-auth-customer)

### CODE REVIEW

File: src/graphql/resolvers/address.resolvers.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+      await service.create(input, context.customer.id)
       return true
     },
   },
   Query: {
-    async addresses() {
-      return service.findAll()
-    }
-  }
+    async addresses(_, __, context) {
```
Comment (mosafa697) [inline src/graphql/resolvers/address.resolvers.ts:13]: what is the using for this: (_, __,)?

Comment (taha7) [reply src/graphql/resolvers/address.resolvers.ts:13]: As we don't need the first 2 parameters, this is telling typescript to not give a warning for the unused vars

**Review by mosafa697 — APPROVED** [review]

File: src/graphql/resolvers/address.resolvers.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
-import { createAddressValidator } from "../../validations/address.validations"
+import { createAddressValidator } from '../../validations/address.validations'
 import service from '../../services/domain/address.service'
 
 export const addressResolvers = {
   Mutation: {
     async createAddress(_, { input }, context) {
       await createAddressValidator.validate(input, { abortEarly: false })
-      await service.create(input)
+      await service.create(input, context.customer.id)
```
Comment (usef) [inline src/graphql/resolvers/address.resolvers.ts:8]: Didn't we agree that we'll use the customer's email?

File: src/graphql/resolvers/address.resolvers.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
       return true
     },
   },
   Query: {
-    async addresses() {
-      return service.findAll()
-    }
-  }
+    async addresses(_, __, context) {
+      return service.findManyBy('customerId', context.customer.id)
```
Comment (usef) [inline src/graphql/resolvers/address.resolvers.ts:14]: Same here

File: src/graphql/resolvers/cart.resolvers.ts [hunk 4]
```diff
@@ -28,8 +28,8 @@ export const cartResolvers = {
     },
   },
   Mutation: {
-    async addToCart(_, { input }) {
-      await service.addToCart(input)
+    async addToCart(_, { input }, context) {
+      await service.addToCart({ ...input, customerId: context.customer.id })
```
Comment (usef) [inline src/graphql/resolvers/cart.resolvers.ts:32]: Same

File: src/services/domain/cart.service.ts [hunk 5]
```diff
@@ -4,12 +4,13 @@ export const addToCart = async ({
   productId,
   variations,
   quantity,
+  customerId
 }: {
   productId: number
   variations: Array<number>
-  quantity: number
+  quantity: number,
+  customerId: number
```
Comment (usef) [inline src/services/domain/cart.service.ts:12]: Same comment

File: src/graphql/resolvers/address.resolvers.ts (hunk 2 above)
Comment (taha7) [reply src/graphql/resolvers/address.resolvers.ts:8]: For orders not for addresses, check [this](https://github.com/ecommerce-cart/backend/pull/6/files#diff-6c6583a13ad05540f06d804e378b6293392e0d6d7a0fe4ba872224a377bbb376L6) comment

**Pull request merged**